nni/algorithms/compression/v2/pytorch/pruning/tools/task_generator.py

```diff
@@ -341,3 +344,149 @@ def generate_tasks(self, task_result: TaskResult) -> List[Task]:
+                if self.current_iteration <= warmup_num:
+                    action = self.agent.random_action()
+                else:
+                    action = self.agent.select_action(self.observation, episode=self.current_iteration)
+                # collect the actions and preserv_idx each layer
+                action, preserv_idx = self.env.step_previous(action)
+                next_action_list.append(action)
+                next_preserve_idx_list.append(preserv_idx)
+
+                if self.env.cur_ind == len(self.env.prunable_idx):
+                    self.env.cur_ind = 0
+                    self.env.visited = [False] * len(self.env.prunable_idx)
+                    done = True
+            
+            # set the reward and mask of last epoch
+            self.env.set_reward(self._tasks[task_result.task_id].score)
+            self.env.set_mask(task_result.pruner_generated_masks)
+
+            if self.env.ini is not False:
+                # self.actions = deepcopy(next_action_list)
+                # self.preserv_idxes = deepcopy(next_preserve_idx_list)
+
+                # use last actions
+                info = None
+                for action, preserv_idx in zip(self.actions, self.preserv_idxes):
+                    observation1, reward, done, info = self.env.step_after(action, preserv_idx)
+                    self.T.append([reward, deepcopy(self.observation), deepcopy(observation1), action, done])
+                    # update
+                    self.observation = deepcopy(observation1)
+                    self.iteration_reward += reward
+
+                self.actions = deepcopy(next_action_list)
+                self.preserv_idxes = deepcopy(next_preserve_idx_list)
+                # end of episode
+                print("Generated config list: {}".format(info['config_list']))
+                _logger.info(
+                    '#%d: iteration_reward: %.4f reward: %.4f, ratio: %.4f\n',
+                        self.current_iteration - 1, self.iteration_reward,
+                        info['reward'],
+                        info['compress_ratio']
+                )
+                final_reward = self.T[-1][0]
+                self._temp_config_list = deepcopy(info['config_list'])
+
+                # agent observe and update policy
+                for _, s_t, s_t1, a_t, done in self.T:
+                    self.agent.observe(final_reward, s_t, s_t1, a_t, done)
+                    if self.current_iteration > warmup_num:
+                        self.agent.update_policy()
+
+                # reset
+                self.observation = None
+                self.iteration_reward = 0.
+                self.T = []
+    
+        
+        
+        # generate task and get task result
+        if self._task_id_candidate < self.total_iteration:
+            task_id = self._task_id_candidate
+            new_config_list = deepcopy(self._temp_config_list)
```

Comment:
if the task is the origin, we need to give a `random_action` config list, seems it is empty right now?

Reply:
This is empty at the origin task. I thought it wouldn't do anything in the initial stage before.
I tried to add `random_action` at origin status, but a conflict seemed to emerge: In every round, we generate a `config_list` and it would be used at next round.
Or I just use a solution which was once used: at the first round, generate a `config_list` and use it right now.
Or just generate a random `config_list` and skip this round?